quarkus/service/src/main/java/org/apache/polaris/service/quarkus/config/QuarkusProducers.java

```diff
@@ -258,6 +265,13 @@ public ActiveRolesProvider activeRolesProvider(
     return activeRolesProviders.select(Identifier.Literal.of(persistenceType)).get();
   }
 
+  @Produces
+  public DatasourceSupplier datasourceSupplier(
+      @Any RelationalJdbcConfiguration relationalJdbcConfiguration,
+      @All List<InstanceHandle<DataSource>> datasources) {
+    return new QuarkusDatasourceSupplier(relationalJdbcConfiguration, datasources);
```

Comment:
This is just doing a regular bean construction injection, so it's likely possible to get rid of this and inject into `QuarkusDatasourceSupplier` directly.